internal/vm/hcs/boot.go

```diff
@@ -0,0 +1,40 @@
+package hcs
+
+import (
+	hcsschema "github.com/Microsoft/hcsshim/internal/hcs/schema2"
+	"github.com/Microsoft/hcsshim/internal/vm"
+	"github.com/Microsoft/hcsshim/osversion"
+	"github.com/pkg/errors"
+)
+
+func (uvm *utilityVM) SetUEFIBoot(dir string, path string, args string) error {
+	if uvm.state != vm.StatePreCreated {
```

Comment:
Should we make setting (and checking) the state atomic or guarded? I was trying to think of if it would even be needed with how we call the lifecycle methods (create, start, terminate etc.) today.

Reply:
If it is necessary (I'm not sure yet) then the entire function that depends on a given state should be guarded against a state change right?

Reply:
Yea good point, we'd want to ensure we're in <state> for the duration of the call and not just reading what state we're in before starting the operation

Reply:
Is this guard really necessary at all? Are we just protecting against programming errors?

An alternative model might be to have a separate Builder interface used up until the thing is created, which then returns the VM type with the runtime methods.

Reply:
Mostly programming error in case this refactor caused any oddities with how things are constructed/called (changes to come). I like the builder idea though and I'll play with that 👍